lib/std/system.nim

```diff
@@ -75,6 +75,11 @@ proc low*[I, T](x: typedesc[array[I, T]]): I {.magic: "Low", noSideEffect.}
 proc high*[T: Ordinal|enum|range](x: typedesc[T]): T {.magic: "High", noSideEffect.}
 proc high*[I, T](x: typedesc[array[I, T]]): I {.magic: "High", noSideEffect.}
 
+proc `[]`*[I, T](x: array[I, T], i: I): T {.magic: "ArrAt".}
+proc `[]`*(x: string, i: int): char {.magic: "StrAt".}
+proc `[]`*(x: cstring, i: int): char {.magic: "CstrAt".}
+proc `[]=`*() {.magic: "ArrPut", varargs.}
```

Comment:
Why is this varargs?

Reply:
Also this one is just:

```nim

proc `[]`*[I, T](x: array[I, T], i: I; elem: sink T) {.inline.} =
  (a[i]) = elem

```

Reply:
Or the same as a `template`, then the `sink` can be omitted too.

Reply:
Made into template for now, I thought maybe we would need a generic version of this (hence varargs for more than 1 subscript argument) but it wouldn't change much from existing Nim which only has this behavior for the builtins anyway